### ISSUE TITLE
Accessing query params from React-router

### ISSUE THREAD

**Issue by shilpan** [makeomatic/redux-connect#17]
Hi,

I've been facing problems accessing the `query` params from the router. The `reduxAsyncConnect` method exposes the `params` argument which has only the declared arguments but not the `query` values. I need them in the static method for server access. How do I do this?

**Comment by AVVS** [comment]
It has everything that router passes into routerProps, I suggest that you log the object you get and find these props or check react-router guides, it should have that information

**Comment by shilpan** [comment]
My route is: 
```javascript
<Route path="filter" component={FilterPage}/>
```

The way I use the static method is:
```javascript
static reduxAsyncConnect(params, store, helpers) {
        console.log(params);
}
```

When I access `/filter?status=true`. The log comes out to be `{}`. But the `props.location` in the `react-router` does populate accordingly.

**Comment by AVVS** [comment]
Static doesnt work that way any longer, please check docs and start using decorator method instead. If you are still confused - let me know

**Comment by AVVS** [comment]
closing for now

**Comment by oyeanuj** [comment]
@AVVS I am facing a similar issue, where I am unable to find the location params either. I am using the decorator version.

Here is my code. I wasn't sure if this is my code issue or an issue with react-router-redux or here but hope you can point me in the right direction.

```javascript
@asyncConnect([{
  promise: ({ store: { dispatch, getState } }) => {

    console.log('in post asyncConnect');
    console.log(getState().routing);
    console.log(params.postSlug);
  }
}])
@connect(
	postContainerSelector,
	mapDispatchToProps
)
export default class PostContainer extends Component {
````

returns 

````javascript
in post asyncConnect
[1] undefined
[1] { locationBeforeTransitions:
[1]    { pathname: '/posts/testing-slugs-123',
[1]      search: '',
[1]      hash: '',
[1]      state: null,
[1]      action: 'POP',
[1]      key: 'kdhr5p',
[1]      query: {},
[1]      '$searchBase': { search: '', searchBase: '' } } }
[1] action @ 17:56:01.278 @reduxAsyncConnect/END_GLOBAL_LOAD
```

Thank you!

**Comment by oyeanuj** [comment]
Ah, I asked too soon. I found the answer here for anyone else coming to look for it in the future. Maybe this should be documented - 

https://github.com/Rezonans/redux-async-connect/issues/53#issuecomment-191527313

Relevant bit: 

```javascript
@asyncConnect([{
  promise: (options) => {
    const {
      store: { dispatch },
      params: { id },
    } = options;

    return dispatch(loadYourDataWidget(id));
  },
}])
```

Here are the full set of properties on that options object passed in.

`store` - commonly accessed props dispatch and getState
`params` - the route params
`helpers`
`matchContext`
`router`
`history`
`location`
`routes`

--

**Comment by sungwoncho** [comment]
@oyeanuj I agree that we should document the list of possible properties for the options. Currently, it seems that [API.md](https://github.com/makeomatic/redux-connect/blob/master/docs/API.MD) has not been updated from the forked version.

@AVVS I would love to help with improve documentation for the options. What do you think?

**Comment by AVVS** [comment]
Sure, I'd merge updated docs quickly

**Comment by floridemai** [comment]
Hi all,

I tried using the `params` options but it seems it is not passed in. Any ideea why? Thanks.
On the server after `loadOnServer`: `<ReduxAsyncConnect {...renderProps} />`
``@asyncConnect([{
  promise: (options) => {
    console.log(options)```
=>
```{ store: 
   { dispatch: [Function],
     subscribe: [Function: subscribe],
     getState: [Function: getState],
     replaceReducer: [Function: replaceReducer] },
  helpers: 
   { client: 
      ApiClient {
        get: [Function],
        post: [Function],
        put: [Function],
        patch: [Function],
        del: [Function] } } }```

**Comment by sungwoncho** [comment]
@floridemai I can confirm `params` options works. Could you post a snippet of how you are using `loadOnServer`?

**Comment by floridemai** [comment]
I found the problem, but thank you. To others having the same problems as me (if there is anyone) it all had to do with `loadOnServer`. I used the redux-async-connect before and i had `loadOnServer({ components: renderProps.components, store, helpers: { client } })` which caused the options to just return the store and the helpers (components are filtered out).

Thanks @sungwoncho for the super quick response.
